cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/Operation.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+/**
+ * Abstract base class to represent Operation.
```

Comment:
This would be a good place to discuss that each operation has a name that must be unique within all operations of the same program's field lineage, and that an optional description can be given. With that, you don't need to document it again for all the subclasses

Reply:
I would also explain here that operations typically have input and/or output fields, and explain that inputs need to be qualified with the name of the operation they originate from, and why, whereas outputs are given as plain field names. This will help users understand the API

Reply:
added javadoc